soh/src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c

```diff
@@ -73,7 +73,7 @@ void BgTreemouth_Init(Actor* thisx, GlobalContext* globalCtx) {
 
     if ((gSaveContext.sceneSetupIndex < 4) && !LINK_IS_ADULT) {
         BgTreemouth_SetupAction(this, func_808BC8B8);
-    } else if (LINK_IS_ADULT || (gSaveContext.sceneSetupIndex == 7)) {
+    } else if (LINK_IS_ADULT && !gSaveContext.n64ddFlag || (gSaveContext.sceneSetupIndex == 7)) {
```

Comment:
for clarity/readability

```suggestion
    } else if ((LINK_IS_ADULT && !gSaveContext.n64ddFlag) || (gSaveContext.sceneSetupIndex == 7)) {
```

Reply:
come do think of it, do we want deku as adult to be a thing in rando without entrance rando? not sure what logic thinks about it

Reply:
Same as fire temple rocks, will add the entrance check here.